### PR TITLE
Open Java modules for Snowflake setup

### LINKED ISSUE #9475: Enable Apache Arrow for improved Snowflake integration
Lack of Arrow is apparently problematic for Snowflake. Enabling it also means we need to add 
```
--add-opens=java.base/java.nio=ALL-UNNAMED
```
as per https://arrow.apache.org/docs/java/install.html

### PR DESCRIPTION
**Pull request by hubertp**
## Pull Request Description

In the absence of other workarounds we need to ensure that every Java process is started with extra options:
`--add-opens=java.base/java.nio=ALL-UNNAMED`
Verified locally. Note: this only affects LS and cli will continue to crash on examples using Snowflake.
Tested manually.

Closes #9475.

### Important Notes

Some workarounds are being considered but we need this change sooner than later.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [X] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by JaroslavTulach — APPROVED** [review]

File: distribution/lib/Standard/Snowflake/0.0.0-dev/src/Snowflake_Details.enso [hunk 1]
```diff
@@ -46,7 +46,7 @@ type Snowflake_Details
     jdbc_properties : Vector (Pair Text Text)
     jdbc_properties self =
         ## Avoid the Arrow dependency (https://community.snowflake.com/s/article/SAP-BW-Java-lang-NoClassDefFoundError-for-Apache-arrow)
-        no_arrow = [Pair.new 'jdbc_query_result_format' 'json']
+        no_arrow = [] #[Pair.new 'jdbc_query_result_format' 'json']
```
Comment (Akirathan) [inline distribution/lib/Standard/Snowflake/0.0.0-dev/src/Snowflake_Details.enso]: Remove this altogether?

**Review by Akirathan — APPROVED** [review]

**Comment by radeusgd** [comment]
> Note: this only affects LS and cli will continue to crash on examples using Snowflake.

I think I don't understand. Does this not affect the launcher as well? If I use `enso run`, I guess that should also use the `default` settings, no?

**Review by radeusgd — APPROVED** [review]

**Comment by JaroslavTulach** [comment]
Btw. the _proper way_ to address the needs of Snowflake JDBC driver is to load each library's JAR files as a separate `ModuleLayer` and use [ModuleLayer.Controller](https://docs.oracle.com/en%2Fjava%2Fjavase%2F21%2Fdocs%2Fapi%2F%2F/java.base/java/lang/ModuleLayer.Controller.html) to open the necessary packages only to Snowflake library.

**Comment by Akirathan** [comment]
> Btw. the _proper way_ to address the needs of Snowflake JDBC driver is to use load each library's JAR files as a separate `ModuleLayer` and use [ModuleLayer.Controller](https://docs.oracle.com/en%2Fjava%2Fjavase%2F21%2Fdocs%2Fapi%2F%2F/java.base/java/lang/ModuleLayer.Controller.html) to open the necessary packages only to Snowflake library.

Tracked in https://github.com/enso-org/enso/issues/7082

**Comment by hubertp** [comment]
> > Note: this only affects LS and cli will continue to crash on examples using Snowflake.
> 
> I think I don't understand. Does this not affect the launcher as well? If I use `enso run`, I guess that should also use the `default` settings, no?

You can always prefix `enso run` with `JAVA_OPTS="--add-opens=java.base/java.nio=ALL-UNNAMED"`.

Having said that, I will add it to the script by default, for convenience.

**Pull request merged**